integration-libs/cds/src/trending-searches/trending-searches.service.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { CdsEndpointsService } from '../services';
+import { CdsConfig } from '../config';
+import { BaseSiteService } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { SearchPhrases } from './trending-searches.component';
+
+const TRENDING_SEARCHES_ENDPOINT_KEY = 'searchIntelligence';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TrendingSearchesService {
+  constructor(
```

Comment:
The same as above, get rid of constructor please and use `inject()` instead. Also check access modifiers to get rid of `private` and use `protected` instead